locator/network_topology_strategy.cc

```diff
@@ -325,16 +325,23 @@ static unsigned calculate_initial_tablets_from_topology(const schema& s, token_m
         }
 
         unsigned tablets_in_dc = rf_in_dc > 0 ? (shards_in_dc + rf_in_dc - 1) / rf_in_dc : 0;
+        if (min_per_shard_tablet_count) {
+            tablets_in_dc = std::max<unsigned>(tablets_in_dc, std::floor(min_per_shard_tablet_count * shards_in_dc));
+        }
         initial_tablets = std::max(initial_tablets, tablets_in_dc);
     }
     rslogger.debug("Estimated {} initial tablets for table {}.{}", initial_tablets, s.ks_name(), s.cf_name());
     return initial_tablets;
 }
 
-future<tablet_map> network_topology_strategy::allocate_tablets_for_new_table(schema_ptr s, token_metadata_ptr tm, unsigned initial_scale) const {
+future<tablet_map> network_topology_strategy::allocate_tablets_for_new_table(schema_ptr s, token_metadata_ptr tm, uint64_t target_tablet_size, std::optional<unsigned> initial_scale) const {
     auto tablet_count = get_initial_tablets();
     if (tablet_count == 0) {
-        tablet_count = calculate_initial_tablets_from_topology(*s, tm, _dc_rep_factor) * initial_scale;
+        tablet_count = tablet_hints().min_tablet_count.value_or(0);
```

Comment:
I think hints should be obtained from "s", since they are per table, not per replication strategy.

Reply:
done in v7